system/lib/pthread/library_pthread_stub.c

```diff
@@ -298,41 +298,69 @@ int pthread_barrier_wait(pthread_barrier_t* mutex) { return 0; }
 /* magic value to track a validly constructed TLS slot */
 #define PTHREAD_TLS_MAGIC_ID 0x02468ACE
```

Comment:
Can you just remove this thing completely now?  It seems that setting the slot to NULL serves the purpose that this used it.

Reply:
Well, before if you used an invalid key it would segfault or fail the magic id test. That is the same in the current code.

I can remove the magic test, and instead put in a range check. That is definitely more solid.

Reply:
Actually I can remove the malloc/free for the object, since its not being referred to externally.

Wow, giving some serious love to a stub function..

Reply:
:)